src/kirin/analysis/callgraph.py

```diff
@@ -22,19 +22,44 @@ class CallGraph(Printable):
     """Mapping from symbol names to methods."""
     backedges: dict[str, set[str]] = field(default_factory=dict)
     """Mapping from symbol names to backedges."""
+    name_counts: dict[str, int] = field(default_factory=dict)
+    """Mapping from symbol names to counts of how many times they have been used."""
+    inv_defs: dict[ir.Method, str] = field(default_factory=dict)
+    """Mapping from symbol names to methods, used for inverse lookups."""
```

Comment:
this looks overly complicated, `ir.Method` implements `__hash__` we can just use `dict[ir.Method, set[ir.Method]]`? or we are missing something?

Reply:
Because you can have methods with different ID's but the same name which was @kaihsin's original problem.